.github/workflows/update-gh-pages.yml

```diff
@@ -10,6 +10,147 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+    - name: Install Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.13'
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+
+        python -m pip install --upgrade pip
+        pip install requests
+
+    - name: Write script
+      shell: bash
+      run: |
+
+        cat > /tmp/catsitemap.py << "EOF"
+        import re
+        import os
+        import xml.etree.ElementTree as ET
+        import requests
+        from xml.dom import minidom
+
+        def download_file(url: str, folder_path: str, file_name: str) -> None:
```

Comment:
Let's accept a url and a dest_path. Also, let's assume this last object to be a [pathlib.Path](https://docs.python.org/3/library/pathlib.html) object. This is the proper way for manipulating file paths with Python.